README.md

```diff
@@ -78,103 +101,366 @@ Modify the app's **android/app/src/main/AndroidManifest.xml** to include the fol
+  enableRealUserMonitoring?: boolean; // Enables Real User Monitoring.
+  disableNetworkMonitoring?: boolean; // If Real User Monitoring is enabled, this disables monitoring network usage.
+  customCrashReportingEndpoint?: string; // Some alternate URL to send crash reports to.
+  customRealUserMonitoringEndpoint?: string; // Some alternate URL to send real user monitoring data to.
+  onBeforeSendingCrashReport?: BeforeSendHandler; // A BeforeSendHandler function (see below).
+  ignoredURLs?: string[]; // An Array of URLs to ignore when network monitoring.
+};
 ```
 
-Initialize the RaygunClient using the `init(RaygunClientOption)` function.
+<br/>
+<br/>
 
+###  BeforeSendHandler 
+The BeforeSendHandler acts as an intermediate function between your application and Raygun. This
+function gets parsed a [CrashReportPayload](#crashreportpayload) and returns a [CrashReportPayload](#crashreportpayload). 
+Before the [CrashReportPayload](#crashreportpayload) is sent to Raygun, this function will interfere 
+with the report. If `null` is returned, then the report gets ignored (not sent to Raygun).
+ 
 ```typescript
-RaygunClient.init(RaygunClientOptions);
+export type BeforeSendHandler = (payload: CrashReportPayload) => CrashReportPayload | null;
 ```
 
-The RaygunClientOptions is constructed with the following fields:
+<br/>
+<br/>
 
-| Options | Required | Default Value | Description |
-| :--- | :------: | :------------ | :---------- |
-| apiKey | false | "" | The key provided with a purchased license of the Raygun Client |
-| version | false | "" | The version of YOUR application |
-| enableCrashReporting | false | false | Enables the usage of Raygun's Crash Reporter |
-| disableNativeCrashReporting | false | false | Given you have crash reporting enabled, this disables Crash Reporting from reporting on errors that occur outside the scope of YOUR program |
-| enableRealUserMonitoring | false | false | Enables the usage of Raygun's Real User Monitoring |
-| disableNetworkMonitoring | false | false | Given you have real user monitoring enabled, this disables the monitor from recording statistics based on Network usage |
-| customCrashReportingEndpoint | false | "" | Sends all Crash Reporting information to an alternative address |
-| customRealUserMonitoringEndpoint | false | "" | Sends all Real User Monitoring information to an alternative address |
-| onBeforeSendingCrashReport | false | null | A handler for crash reports before they are sent (more below) |
-| ignoredURLs | false | [] | An array of URLs to ignore. This is utilized with network monitoring, which will be useless without enabling Real User Monitoring (and leaving Network Monitoring enabled) |
+###  Breadcrumb 
+A Breadcrumb is a container for simple pieces of information that are used to aid with identifying
+issues. These are sent away with a [CrashReportPayload](#crashreportpayload).
```

Comment:
Maybe we can include links our documentation which describes what breadcrumbs are in more detail alongside some examples? (If we have this documentation).

Reply:
Do you know where to find these links? I haven't managed to find any yet.

IN PROGRESS

Reply:
https://raygun.com/documentation/product-guides/crash-reporting/breadcrumbs/